caasoperator connecting to API server

mikemccracken opened the pull request:
api-caller is currently failing when it attempts to use the Agent facade to do the password changing dance.

Commit:
Add one-command pod bounce

Hacky, this assumes there's only the one juju-operator-* pod running,
and it still requires pushing to docker hub

Signed-off-by: Michael McCracken <mike.mccracken@canonical.com>

Commit:
Un-comment API caller setup

Signed-off-by: Michael McCracken <mike.mccracken@canonical.com>

Commit:
Add Agent Facade to caas model facades

Signed-off-by: Michael McCracken <mike.mccracken@canonical.com>

mikemccracken commented:
WIP - not ready to merge necessarily

mjs commented on cmd/jujud/agent/caasoperator/manifolds.go:
```diff
@@ -164,18 +164,18 @@ func Manifolds(config ManifoldsConfig) dependency.Manifolds {
-		// 	AgentName:     agentName,
-		// 	APICallerName: apiCallerName,
-		// })),
+		loggingConfigUpdaterName: ifNotMigrating(logger.Manifold(logger.ManifoldConfig{
```
Does `ifNotMigrating` even work for a CAAS model? I think that should go. We certainly don't care about migrations for the prototype.

Commit:
Remove ifNotMigrating for caasoperator

also removes definition so we notice if we uncomment something and add
it back in

Signed-off-by: Michael McCracken <mike.mccracken@canonical.com>

mikemccracken commented:
ok to merge
